lib/irb/input-method.rb

```diff
@@ -270,7 +270,13 @@ def initialize(completor)
           proc do |output, complete: |
             next unless IRB::Color.colorable?
             lvars = IRB.CurrentContext&.local_variables || []
-            IRB::Color.colorize_code(output, complete: complete, local_variables: lvars)
+            if IRB.CurrentContext&.irb&.command?(output)
```

Comment:
Since `command?` calls `parse_command` underneath, I think we can directly call `parse_command` to get a more constructive `Statement` object here instead?

(I also opened #993 to move `parse_command` to `Context`).

Reply:
I wasn't clear (and also not completely correct) about the idea: maybe we can use `Irb#build_statement` to get a `Statement` back, and put most of the block into a method like `Statement#colorized_output`.

BUT, I think it may actually be a wrong coupling to make `Statement` handle display logic as well + it now needs to take in local variables too. So let's drop it 🙈